addons/survey/tests/test_survey.py

```diff
@@ -1,13 +1,83 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import _
+from freezegun import freeze_time
+
+from odoo import _, fields
 from odoo.addons.survey.tests import common
 from odoo.tests.common import users
 
 
 class TestSurveyInternals(common.TestSurveyCommon):
 
+    def test_answer_attempts_count(self):
+        """ As 'attempts_number' and 'attempts_count' are computed using raw SQL queries, let us
+        test the results. """
+
+        test_survey = self.env['survey.survey'].create({
+            'title': 'Test Survey',
+            'is_attempts_limited': True,
+            'attempts_limit': 4
+        })
+
+        first_attempt = self._add_answer(test_survey, self.survey_user.partner_id, **{'state': 'done'})
+        second_attempt = self._add_answer(test_survey, self.survey_user.partner_id, **{'state': 'done'})
+        third_attempt = self._add_answer(test_survey, self.survey_user.partner_id, **{'state': 'done'})
+        fourth_attempt = self._add_answer(test_survey, self.survey_user.partner_id, **{'state': 'done'})
+
```

Comment:
Currently, the computed method will be called 4 times (because we read each value individually)

It could be nice to add this line, so the computed method is called in batch (so we test it work well in batch as well)
```python
(first_attempt | second_attempt | third_attempt | fourth_attempt).mapped('attempts_number')
```

(maybe there's a prettier way to do it)

What do you think ?